apiserver/pkg/http/client.go

```diff
@@ -20,6 +20,9 @@ type KuberayAPIServerClient struct {
 	marshaler   *protojson.MarshalOptions
 	unmarshaler *protojson.UnmarshalOptions
 	baseURL     string
+
+	// Store http request handling function for unit test purpose.
```

Comment:
why not define an interface instead? Which way is more common cc @dentiny @rueian?

```go
type APIServerClient interface {
    ExecuteHttpRequest(...) ...
}
```

Reply:
Actually, using the standard http.RoundTripper is more common in this case.

```go
type RoundTripperFunc func(request *http.Request) (*http.Response, error)

func (fn RoundTripperFunc) RoundTrip(request *http.Request) (*http.Response, error) {
	return fn(request)
}

func MockNewKuberayAPIServerClient(rt RoundTripperFunc) *KuberayAPIServerClient {
	return NewKuberayAPIServerClient("", &http.Client{
		Transport: rt,
	})
}

func TestXXX() {
	clinet := MockNewKuberayAPIServerClient(func(request *http.Request) (*http.Response, error) {
		return &http.Response{Body: io.NopCloser(strings.NewReader("oooo"))},nil
	})
}
```

Reply:
> why not define an interface instead? Which way is more common

interface has performance impact, because interface involves heap allocation for sure
I don't see any reference recommendation, let me know if you know any :)

Reply:
> Actually, using the standard http.RoundTripper is more common in this case.

If you return `http.Response`, you need to squeeze `ListClustersResponse` (our protobuf message) into http response, which is more complicated than hijacking `executeHttpRequest`.

Reply:
Discussed offline, leave a TODO to revisit later.